bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Journal.java

```diff
@@ -901,7 +921,7 @@ public void run() {
                                 TimeUnit.NANOSECONDS);
                     }
 
-                    if (toFlush.isEmpty()) {
+                    if (numEntriesToFlush == 0) {
```

Comment:
I don't think we need to change this line here. It is better to use toFlush here, because toFlush will never decrease on each run, while `numEntriesToFlush` can be changed. it is good to use toFlush as the condition, because all the group commit logic is based on toFlush.

Reply:
ok, fine